src/components/Auth/components/RegisterForm/RegisterForm.spec.jsx

```diff
@@ -0,0 +1,112 @@
+/* eslint no-console: 0  */
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import { spy } from 'sinon';
+
+// Unwrapped Component
+import { RegisterForm } from './RegisterForm.jsx';
+
+describe('RegisterForm.jsx', () => {
+  let props;
+  let wrapper;
+  let handleRegisterSpy;
+  let validateSpy;
+
+  beforeEach(() => {
+    handleRegisterSpy = spy();
+    validateSpy = spy();
+
+    props = {
+      handleRegister: handleRegisterSpy,
+      regErr: '',
+      waiting: false,
+      classes: {},
+      emailErr: '',
+      passErr: '',
+      confirmErr: '',
+      validate: validateSpy,
+    };
+
+    wrapper = shallow(<RegisterForm {...props} />);
+  });
+
+  it('should not allow the user to submit the form at render', () => wrapper.instance().checkForErrors()
+    .catch((err) => {
+      expect(err).to.exist;
+    }));
+
+  it('should not allow the form to be submitted if there are errors or if we are waiting', () => {
+    const propsWithErrors = {
+      ...props,
+      emailErr: 'Bad Email!',
+      passErr: 'Bad Password!',
+    };
+    wrapper.setProps(propsWithErrors);
+    return wrapper.instance().checkForErrors()
+      .catch((err) => {
+        expect(err).to.exist;
+      });
+  });
+
+  it('should allow the form to be submitted if errors are cleared', () => {
+    const propsWithoutErrors = {
+      ...props,
+    };
+
+    const filledForm = {
+      email: 'registration.form@ttu.edu',
+      password: 'Password89067!',
+      confirmPass: 'Password89067!',
+      name: 'Student',
+      role: 'student',
+    };
+
+    // Mock a change in input since the input components are tested seperately
+    wrapper.setProps(propsWithoutErrors);
+    wrapper.setState(filledForm);
+    return wrapper.instance().checkForErrors()
+      .then(() => {
+        expect(true).to.equal(true);
+      });
+  });
+
+  it('should call validate() and checkForErrors() when input is recieved', () => {
+    const testState = {
+      email: 'newEmail',
+      password: 'newPasswordThatSucks',
+      confirmPass: 'thisWillThrowAnError',
+      name: 'Nils',
+      role: 'BDFL',
+      preventSubmit: false,
+    };
+    const changePasswordEvent = {
+      target: { value: 'AnotherPassword' },
+    };
+    wrapper.setState(testState);
+    wrapper.instance().handleInputChange('password')(changePasswordEvent);
+    expect(props.validate.called).to.equal(true);
```

Comment:
Does this imply calling `checkForErrors`?

Reply:
Yeah oops... I'm not sure what was going through my mind when i was writing these test. I'll fix this now